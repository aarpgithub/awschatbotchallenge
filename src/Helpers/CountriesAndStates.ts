import states = require("us-state-codes");

export const stateNameToCode = (state: string) => {
  let stateCode = states.sanitizeStateName(state);
  if (stateCode == null) {
    throw Error(`Error missing field state.  Input: ${state}`);
  }
  stateCode = states.getStateCodeByStateName(stateCode);
  if (stateCode == null) {
    throw Error("Error converting state to code.  Input: ${state}");
  }
  return stateCode;
};
