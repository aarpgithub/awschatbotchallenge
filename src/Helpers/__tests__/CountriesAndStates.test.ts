import { stateNameToCode } from "../CountriesAndStates";

it("Should produce the correct abbreviation for example state names", () => {
  const testCases = ["California", "Nevada", "Washington"];
  const codes = testCases.map(e => {
    return stateNameToCode(e);
  });
  expect(codes).toMatchSnapshot();
});

it("Should throw an error on invalid state names", () => {
  const shouldThrow = () => {
    return stateNameToCode("Kalifornia");
  };

  expect(shouldThrow).toThrowErrorMatchingSnapshot();
});
