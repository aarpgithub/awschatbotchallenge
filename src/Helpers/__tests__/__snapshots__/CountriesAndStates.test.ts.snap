// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should produce the correct abbreviation for example state names 1`] = `
Array [
  "CA",
  "NV",
  "WA",
]
`;

exports[`Should throw an error on invalid state names 1`] = `"Error missing field state.  Input: Kalifornia"`;
